internal/ingress/controller/endpoints.go

```diff
@@ -50,6 +50,12 @@ func getEndpoints(s *corev1.Service, port *corev1.ServicePort, proto corev1.Prot
 
 	// ExternalName services
 	if s.Spec.Type == corev1.ServiceTypeExternalName {
+		if ip := net.ParseIP(s.Spec.ExternalName); s.Spec.ExternalName == "localhost" ||
```

Comment:
It may not be necessary to determine whether it is `localhost`.

external name is the CNAME record of DNS, which allows multiple layers of recursion

Reply:
https://github.com/kubernetes/ingress-nginx/pull/7092/files#diff-de8de16ab07e797b0d2e2d81dde711cfb2a34e1ddaf501c34363ebe0aa9516e6R224 removed the `GET` method, so I think it's ok.

Reply:
@tao12345666333 thanks :) Yes, but the problem is the way ingress interprets this.

I think you are right if I create something like a CNAME in my own DNS called mylocal.mydomain.com and point it to 127.0.0.1 and then point this to externalName BUT as @moonming said we are also removing the GET method right now.

In a future, we can try to reconcile the CNAME and check if it points to a localhost ip as well, but this might be pretty 'annoying' to solve on Go and sync with Lua.

Maybe, we can add this verification in Lua or NGINX later: if the endpoint is 127.0.0.1, then drop it.

Reply:
> @tao12345666333 thanks :) Yes, but the problem is the way ingress interprets this.
> 
> 
> 
> I think you are right if I create something like a CNAME in my own DNS called mylocal.mydomain.com and point it to 127.0.0.1 and then point this to externalName BUT as @moonming said we are also removing the GET method right now.
> 
> 
> 
> In a future, we can try to reconcile the CNAME and check if it points to a localhost ip as well, but this might be pretty 'annoying' to solve on Go and sync with Lua.
> 
> 
> 
> Maybe, we can add this verification in Lua or NGINX later: if the endpoint is 127.0.0.1, then drop it.

sounds good to me